tests/llm/test_fallback_integration.py

```
@@ -0,0 +1,198 @@
+            return primary_backend
+        elif provider == "lmstudio":
+            return fallback_backend
+        raise ValueError(f"Unexpected provider: {provider}")
+
+    mock_client = MagicMock()
+
+    with (
+        patch("src.llm.runtime.client_for_model_config", return_value=mock_client),
+        patch.object(executor_mod, "backend_for_provider", side_effect=mock_backend_for_provider),
+    ):
+        with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)